src/StoreApi/Utilities/JsonWebToken.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+namespace Automattic\WooCommerce\StoreApi\Utilities;
+
+/**
+ * JsonWebToken class.
+ *
+ * Simple Json Web Token generator & verifier static utility class, currently supporting only HS256 signatures.
+ */
+final class JsonWebToken {
+
+	/**
+	 * JWT header type.
+	 *
+	 * @var string
+	 */
+	private static $type = 'JWT';
+
+	/**
+	 * JWT algorithm to generate signature.
+	 *
+	 * @var string
+	 */
+	private static $algorithm = 'HS256';
+
+	/**
+	 * Generates a token from provided data and secret.
+	 *
+	 * @param array  $payload Payload data.
+	 * @param string $secret The secret used to generate the signature.
+	 *
+	 * @return string
+	 */
+	public static function create( array $payload, string $secret ) {
+		$header    = self::to_base_64_url( self::generate_header() );
+		$payload   = self::to_base_64_url( self::generate_payload( $payload ) );
+		$signature = self::to_base_64_url( self::generate_signature( $header . '.' . $payload, $secret ) );
+
+		return $header . '.' . $payload . '.' . $signature;
+	}
+
+	/**
+	 * Validates a provided token against the provided secret.
+	 * Checks for format, valid header for our class, expiration claim validity and signature.
+	 * https://datatracker.ietf.org/doc/html/rfc7519#section-7.2
+	 *
+	 * @param string $token Full token string.
+	 * @param string $secret The secret used to generate the signature.
+	 *
+	 * @return bool
+	 */
+	public static function validate( string $token, string $secret ) {
+		/**
+		 * Confirm the structure of a JSON Web Token, it has three parts separated
+		 * by dots and complies with Base64URL standards.
+		 */
+		if ( preg_match( '/^[a-zA-Z\d\-_=]+\.[a-zA-Z\d\-_=]+\.[a-zA-Z\d\-_=]+$/', $token ) !== 1 ) {
+			return false;
+		}
+
+		$parts = self::get_parts( $token );
+
+		/**
+		 * Check if header declares a supported JWT by this class.
+		 */
+		if (
+			! is_object( $parts->header ) ||
+			! property_exists( $parts->header, 'typ' ) ||
+			! property_exists( $parts->header, 'alg' ) ||
```

Comment:
Alternatively could combine these into a single isset if cleaner. e.g. `! isset( $parts->header->typ, $parts->header->alg )`

Reply:
I usually tend to use `isset()` for particular cases since it also returns false if the property existed but had `null` value. We couldn't compare it to `null` later (won't probably ever be the case here). Personally, I also find these two lines more legible and semantic "an object possessing these properties carrying these values".

What do you think?